File: social-service/src/main/java/com/eirsteir/coffeewithme/social/service/coffeebreak/CoffeeBreakService.java
package com.eirsteir.coffeewithme.social.service.coffeebreak;

import com.eirsteir.coffeewithme.commons.domain.coffeebreak.CoffeeBreakDetails;
import com.eirsteir.coffeewithme.commons.security.UserDetailsImpl;
import com.eirsteir.coffeewithme.social.web.request.CoffeeBreakRequest;

public interface CoffeeBreakService {

  CoffeeBreakDetails registerCoffeeBreak(CoffeeBreakRequest coffeeBreakDto, UserDetailsImpl principal);

}

File: social-service/src/main/kotlin/com/eirsteir/coffeewithme/social/service/coffeebreak/CoffeeBreakService.kt
package com.eirsteir.coffeewithme.social.service.coffeebreak

import com.eirsteir.coffeewithme.commons.domain.coffeebreak.CoffeeBreakDetails
import com.eirsteir.coffeewithme.commons.security.UserDetailsImpl
import com.eirsteir.coffeewithme.social.web.request.CoffeeBreakRequest

interface CoffeeBreakService {
    fun registerCoffeeBreak(
        coffeeBreakRequest: CoffeeBreakRequest, currentUser: UserDetailsImpl
    ): CoffeeBreakDetails
}